ui-v2/app/utils/ticker/index.js

```diff
@@ -0,0 +1,209 @@
+  }
+  removeEventListener() {
+    return this.getTicker().removeEventListener(...arguments);
+  }
+  stop() {
+    return this.gotoAndStop(this._currentframe);
+  }
+  play() {
+    return this.gotoAndPlay(this._currentframe);
+  }
+  start() {
+    return this.gotoAndPlay(this._currentframe);
+  }
+  gotoAndStop(frame) {
+    this._currentframe = frame;
+    const ticker = this.getTicker();
+    if (ticker.isAlive()) {
+      ticker.stop();
+    }
+    return this;
+  }
+  gotoAndPlay(frame) {
+    this._currentframe = frame;
+    const ticker = this.getTicker();
+    if (!ticker.isAlive()) {
+      ticker.start();
+    }
+    return this;
+  }
+  getTicker() {
+    if (typeof this._ticker === 'undefined') {
+      this._ticker = new Ticker(this);
+    }
+    return this._ticker;
+  }
+  setFrames(frames) {
+    this._totalframes = frames;
+    return this;
+  }
+  setIncrement(inc) {
+    this._increment = inc;
+    return this;
+  }
+};
+const Cubic = {
+  easeOut: function(t, b, c, d) {
+    t /= d;
+    t--;
+    return c * (t * t * t + 1) + b;
+  },
+};
+const TickerGroup = IntervalTickerGroup;
+export const Tween = class extends TimelineAbstract {
+  static destroy() {
+    Ticker.destroy();
+  }
+  static to(start, finish, frames, method) {
+    Object.keys(finish).forEach(function(key) {
+      finish[key] -= start[key];
+    });
+    return new Tween(start, finish, frames, method).play();
+  }
+  constructor(obj, props, frames = 12, method = Cubic.easeOut) {
+    super();
+    this.setMethod(method);
+    this.setProps(props);
+    this.setTarget(obj);
+    this.setFrames(frames);
+    this.tick = this.forwards;
+  }
+  _process() {
+    Object.keys(this._props).forEach(key => {
+      const num = this._method(
+        this._currentframe,
+        this._initialstate[key],
+        this._props[key],
+        this._totalframes
+      );
+      // this._target[key] = num;
+      set(this._target, key, num);
+    });
+  }
+  forwards() {
+    if (this._currentframe <= this._totalframes) {
+      this._process();
+      this._currentframe += this._increment;
+    } else {
+      this._currentframe = this._totalframes;
+      this.getTicker().stop();
+    }
+  }
+  backwards() {
+    this._currentframe -= this._increment;
+    if (this._currentframe >= 0) {
+      this._process();
+    } else {
+      this.run = this.forwards;
+      this._currentframe = 1;
+      this.getTicker().stop();
+    }
+  }
+  gotoAndPlay() {
+    if (typeof this._initialstate === 'undefined') {
+      this._initialstate = {};
+      Object.keys(this._props).forEach(key => {
+        this._initialstate[key] = this._target[key];
+      });
+    }
+    return super.gotoAndPlay(...arguments);
+  }
+  setTarget(target) {
+    this._target = target;
+  }
+  getTarget(target) {
+    return this._target;
+  }
+  setProps(props) {
+    this._props = props;
+    return this;
+  }
+  setMethod(method) {
+    this._method = method;
+  }
+};
```

Comment:
Just a quick note here ^ I'll be splitting all this up into separate files at some point